libraries/AP_HAL_ChibiOS/hwdef/PixC4-Jetson/hwdef.dat

```diff
@@ -0,0 +1,326 @@
+PD14 TIM4_CH3 TIM4 PWM(6) GPIO(55)
+# we need to disable DMA on the last 2 FMU channels
+# as timer 12 doesn't have a TIMn_UP DMA option
+PH6  TIM12_CH1 TIM12 PWM(7) GPIO(56) NODMA
+PH9  TIM12_CH2 TIM12 PWM(8) GPIO(57) NODMA
+
+define BOARD_PWM_COUNT_DEFAULT 8
+
+# PWM output for buzzer
+PE5 TIM15_CH1 TIM15 GPIO(77) ALARM
+
+# analog in
+PA0 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+PA1 BATT_CURRENT_SENS ADC1 SCALE(1)
+
+PA2 BATT2_VOLTAGE_SENS ADC1 SCALE(1)
+PA3 BATT2_CURRENT_SENS ADC1 SCALE(1)
+
+PC4 SPARE1_ADC1 ADC1 SCALE(1)
+PA4 SPARE2_ADC1 ADC1 SCALE(1)
+
+PB0 RSSI_IN ADC1 SCALE(1)
+
+PC0 VDD_5V_SENS ADC1 SCALE(2)
+PC1 SCALED_V3V3 ADC1 SCALE(2)
+
+# setup scaling defaults
+define HAL_BATT_VOLT_SCALE 18.0
+define HAL_BATT_CURR_SCALE 24.0
+define HAL_BATT_VOLT_PIN 0
+define HAL_BATT_CURR_PIN 1
+define HAL_BATT2_VOLT_PIN 2
+define HAL_BATT2_CURR_PIN 3
+
+# CAN bus
+PI9  CAN1_RX CAN1
+PH13 CAN1_TX CAN1
+
+PB12 CAN2_RX CAN2
+PB13 CAN2_TX CAN2
+
+PH2 GPIO_CAN1_SILENT OUTPUT PUSHPULL SPEED_LOW LOW GPIO(70)
+PH3 GPIO_CAN2_SILENT OUTPUT PUSHPULL SPEED_LOW LOW GPIO(71)
+PH4 GPIO_CAN3_SILENT OUTPUT PUSHPULL SPEED_LOW LOW GPIO(72)
+
+# GPIOs
+PA7 HEATER_EN OUTPUT LOW GPIO(80)
+define HAL_HEATER_GPIO_PIN 80
+
+PG1 VDD_BRICK_VALID INPUT PULLUP
+PG2 VDD_BRICK2_VALID INPUT PULLUP
+PG3 nVBUS INPUT PULLDOWN
+PA9 VBUS INPUT
+PF13 VDD_5V_HIPOWER_OC INPUT PULLUP
+PE15 VDD_5V_PERIPH_OC INPUT PULLUP
+
+# capture pins
+PA5 FMU_CAP1 INPUT  GPIO(58)
+PB3 FMU_CAP2 INPUT  GPIO(59)
+PB11 FMU_CAP3 INPUT GPIO(60)
+PI0 FMU_SPARE_4 INPUT GPIO(61)
+
+# SPI devices
+# note that SPI6 is disabled by default, try to enable it once we get one sensor up
+SPIDEV ms5611	       SPI4 DEVID5  BARO_CS    MODE3  5*MHZ  5*MHZ
+SPIDEV icm20602        SPI1 DEVID1  ICM20602_CS  MODE3  2*MHZ  8*MHZ
+SPIDEV icm20602_2      SPI1 DEVID2  ICM20602_CS2  MODE3  2*MHZ  8*MHZ
+SPIDEV icm42688        SPI1 DEVID3  ICM42688_CS  MODE3  2*MHZ  8*MHZ
+SPIDEV ramtron         SPI2 DEVID1  FRAM_CS      MODE3  8*MHZ  8*MHZ 
+
+# 3 IMUs
+IMU Invensense SPI:icm20602 ROTATION_YAW_90
+IMU Invensense SPI:icm20602_2 ROTATION_YAW_90
+IMU Invensensev3 SPI:icm42688 ROTATION_PITCH_180
+
+define HAL_DEFAULT_INS_FAST_SAMPLE 1
+
+# probe external I2C compasses plus some internal IST8310
+# we also probe some external IST8310 with a non-standard orientation
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+COMPASS RM3100 I2C:0:0x20 false ROTATION_NONE
+
+# baro
+BARO MS56XX      SPI:ms5611
+
+# microSD support
+PC8 SDMMC1_D0 SDMMC1
+PC9 SDMMC1_D1 SDMMC1
+PC10 SDMMC1_D2 SDMMC1
+PC11 SDMMC1_D3 SDMMC1
+PC12 SDMMC1_CK SDMMC1
+PD2 SDMMC1_CMD SDMMC1
+
+# red LED marked as B/E
+PB1 LED_RED OUTPUT OPENDRAIN GPIO(90)
+
+# green LED of FMU RGB LED
+PC6 LED_GREEN OUTPUT GPIO(91) LOW
+
+# blue LED of FMU RGB LED
+PC7 LED_BLUE OUTPUT GPIO(92) HIGH
+
+# setup for Board LEDs
+define HAL_GPIO_A_LED_PIN 90
+define HAL_GPIO_B_LED_PIN 92
+define HAL_GPIO_LED_ON 0
+
+# enable RAMTROM parameter storage
+define HAL_STORAGE_SIZE 16384
```

Comment:
are you sure it's only 16k? most modern boards have 32k, and can do a lot more waypoints

Reply:
updated to 32k